### PR TITLE
Lease time: Switch from ticks to milliseconds without fractions and add minimum

### PR DESCRIPTION
**Pull request by danielmarbach**
Alternative to #839

### CODE REVIEW

File: src/NServiceBus.RavenDB/SagaPersister/SagaPersister.cs [hunk 1]
```diff
@@ -19,7 +19,8 @@ public SagaPersister(SagaPersistenceConfiguration options, bool useClusterWideTr
             this.useClusterWideTransactions = useClusterWideTransactions;
             leaseLockTime = options.LeaseLockTime;
             enablePessimisticLocking = options.EnablePessimisticLocking;
-            acquireLeaseLockRefreshMaximumDelayTicks = (int)options.LeaseLockAcquisitionMaximumRefreshDelay.Ticks;
+            acquireLeaseLockRefreshMaximumDelayMillisecondsWithoutFractions = Convert.ToInt32(options.LeaseLockAcquisitionMaximumRefreshDelay.TotalMilliseconds);
+            acquireLeaseLockRefreshMinimumDelayMillisecondsWithoutFractions = Convert.ToInt32(TimeSpan.FromMilliseconds(5).TotalMilliseconds);
```
Comment (ramonsmits) [inline src/NServiceBus.RavenDB/SagaPersister/SagaPersister.cs:23]:
```suggestion
            acquireLeaseLockRefreshMinimumDelayMillisecondsWithoutFractions = 5;
```

File: src/NServiceBus.RavenDB/SagaPersister/SagaPersister.cs [hunk 2]
```diff
@@ -245,7 +246,8 @@ static string DocumentIdForSagaData(IAsyncDocumentSession documentSession, Type
         static readonly Random random = new Random();
 
         readonly bool enablePessimisticLocking;
-        readonly int acquireLeaseLockRefreshMaximumDelayTicks;
+        readonly int acquireLeaseLockRefreshMaximumDelayMillisecondsWithoutFractions;
+        readonly int acquireLeaseLockRefreshMinimumDelayMillisecondsWithoutFractions;
```
Comment (ramonsmits) [inline src/NServiceBus.RavenDB/SagaPersister/SagaPersister.cs:250]: Personally the `WithoutFractions` isn't needed for my taste, that is obvious based on the `int` type.
```suggestion
        readonly int acquireLeaseLockRefreshMaximumDelayMilliseconds;
        readonly int acquireLeaseLockRefreshMinimumDelayMilliseconds;
```

**Review by ramonsmits — APPROVED** [review]
Personally like this better. Milliseconds usually are easier to understand compared to ticks.

File: src/NServiceBus.RavenDB/SagaPersister/SagaPersister.cs [hunk 3]
```diff
@@ -208,7 +209,7 @@ async Task<long> AcquireLease(IDocumentStore store, string sagaDataDocId, Cancel
                             }
                         }
 
-                        await Task.Delay(TimeSpan.FromTicks(5 + random.Next(acquireLeaseLockRefreshMaximumDelayTicks)), token).ConfigureAwait(false);
+                        await Task.Delay(TimeSpan.FromMilliseconds(random.Next(acquireLeaseLockRefreshMinimumDelayMillisecondsWithoutFractions, acquireLeaseLockRefreshMaximumDelayMillisecondsWithoutFractions)), token).ConfigureAwait(false);
```
Comment (danielmarbach) [inline src/NServiceBus.RavenDB/SagaPersister/SagaPersister.cs:212]:
```suggestion
                        await Task.Delay(TimeSpan.FromMilliseconds(random.Next(acquireLeaseLockRefreshMinimumDelayMilliseconds, acquireLeaseLockRefreshMaximumDelayMilliseconds)), token).ConfigureAwait(false);
```

**Pull request merged**